src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts

```diff
@@ -62,8 +62,10 @@ export function useSetupTradeState(): void {
   const { state, updateState } = useTradeState()
 
   const chainIdFromUrl = tradeStateFromUrl.chainId
-  // When there is no account from provider, then we consider provider as not connected and use chainId from URL as current
-  const currentChainId = (account ? providerChainId : chainIdFromUrl) || providerChainId
+  // When there is no account from provider (wallet is not connected)
+  // And there is no chainId in provider (edge case in E2E tests)
+  // Then we use chainId from URL as current
+  const currentChainId = (account ? providerChainId : providerChainId || chainIdFromUrl) || providerChainId
```

Comment:
A bit of a nitpick however why not do this in multiple steps for readability? Something along the lines of;

```
let currentChainId = providerChainId;

if (!account && !currentChainId) {
  currentChainId = chainIdFromUrl;
}
```

Since you want only in one case to use `chainIdFromUrl` and everything else seems to be falling back to `providerChainId`.

Reply:
Thank you!
I don't like using `let` :) Refactored by another way

Reply:
Hahah fair enough! Yours is better anyways 🫡